core/frontend/src/components/wifi/WifiUpdater.vue

```diff
@@ -0,0 +1,129 @@
+
+          if (response.data.wpa_state != 'COMPLETED') {
+            wifi_store.setCurrentNetwork(null)
+            return
+          }
+
+          const scanned_network = wifi_store.available_networks.find(
+            (network) => network.ssid === response.data.ssid,
+          )
+          const saved_network = wifi_store.saved_networks.find(
+            (network) => network.ssid === response.data.ssid,
+          )
+
+          wifi_store.setCurrentNetwork({
+            ssid: response.data.ssid,
+            signal: scanned_network ? scanned_network.signal : 0,
+            locked: response.data.key_mgmt.includes('WPA'),
+            saved: saved_network ? true : false,
+          })
+        })
+        .catch((error) => {
+          common_store.openError({
+            index: ErrorIndex.WifiStatusFail,
+            message: `Could not fetch wifi status: ${error.message}`,
+          })
+          wifi_store.setCurrentNetwork(null)
+        })
+    },
+    async fetchAvailableNetworks (): Promise<void> {
+      await axios({
+        method: 'get',
+        url: `${wifi_store.API_URL}/scan`,
+        timeout: 15000,
+      })
+        .then((response) => {
+          common_store.closeError(ErrorIndex.WifiScanFail)
+
+          const saved_networks_ssids = wifi_store.saved_networks.map((network: SavedNetwork) => network['ssid'])
+          const available_networks = response.data.map((network: WPANetwork) => {
+            return {
+              ssid: network.ssid,
+              signal: network.signallevel,
+              locked: network.flags.includes('WPA'),
+              saved: saved_networks_ssids.includes(network.ssid),
+            }
+          })
+          wifi_store.setAvailableNetworks(available_networks)
+        })
+        .catch((error) => {
+          common_store.openError({
+            index: ErrorIndex.WifiScanFail,
+            message: `Could not scan wifi networks: ${error.message}`,
+          })
+          wifi_store.setAvailableNetworks([])
+        })
+    },
+    async fetchSavedNetworks (): Promise<void> {
+      await axios({
+        method: 'get',
+        url: `${wifi_store.API_URL}/saved`,
+        timeout: 1000,
+      })
+        .then((response) => {
+          common_store.closeError(ErrorIndex.WifiSavedFail)
+          wifi_store.setSavedNetworks(response.data)
+        })
+        .catch((error) => {
+          common_store.openError({
+            index: ErrorIndex.WifiSavedFail,
+            message: `Could not fetch saved networks: ${error.message}.`,
+          })
+          wifi_store.setSavedNetworks([])
+        })
+    },
+    sleep (ms: number): Promise<void> {
+      return new Promise((resolve) => setTimeout(resolve, ms))
+    },
+    async updateNetworkStatus (): Promise<void> {
+      await this.fetchNetworkStatus()
+      await this.sleep(3000)
+      this.updateNetworkStatus()
+    },
+    async updateSavedNetworks (): Promise<void> {
```

Comment:
I don't think we need to constantly fetch the saved and available networks.

Reply:
Didn't get the idea, exactly. You mean to just fetch at boot?

Reply:
I still haven't run it here, but I got the impression that this is always running.
we should have it scanning only when the wifi widget is open.

Reply:
It's an option, for sure.
Was like that before, in fact, but I changed.